addons/mail/models/mail_thread.py

```diff
@@ -3436,16 +3436,16 @@ def message_unsubscribe(self, partner_ids=None):
         # not necessary for computation, but saves an access right check
         if not partner_ids:
             return True
-        if set(partner_ids) == set([self.env.user.partner_id.id]):
-            self.check_access_rights('read')
-            self.check_access_rule('read')
-        else:
+        if set(partner_ids) != {self.env.user.partner_id.id}:
```

Comment:
I remember we talked about this, but don't remember all details, should probably deserves a comment :D